clients/src/main/java/org/apache/kafka/common/record/RecordBatch.java

```diff
@@ -151,7 +152,7 @@
     boolean hasProducerId();
 
     /**
-     * Get the first sequence number of this record batch.
+     * Get the base sequence number of this record batch. Note that this value is not impacted by log cleaning.
```

Comment:
I think your update did not include `lastSequence`, I'll do that before merging.